crates/sargon-uniffi/src/signing/subintent/sign_request.rs

```diff
@@ -0,0 +1,75 @@
+use crate::prelude::*;
+use sargon::SignRequest as InternalSignRequest;
+use sargon::Subintent as InternalSubintent;
+
+#[derive(Clone, PartialEq, Eq, uniffi::Record)]
+pub struct SignRequestForSubintent {
+    pub factor_source_kind: FactorSourceKind,
+
+    /// Per factor source, a set of transactions to sign, with
+    /// multiple derivations paths.
+    pub per_factor_source: HashMap<
```

Comment:
oh! Be mindful of order!!! `HashMap` distorts the order! and I think we MUST retain the order! So create a UniFFI exported tuple for the key-value-pair and use `Vec`.

Reply:
Right wanted to ask about these.